src/backend/distributed/operations/auto_shard_split.c

```diff
@@ -0,0 +1,141 @@
+#include "postgres.h"
+#include "libpq-fe.h"
+#include <math.h>
+#include "distributed/pg_version_constants.h"
+#include "access/htup_details.h"
+#include "access/genam.h"
+#include "catalog/pg_type.h"
+#include "catalog/pg_proc.h"
+#include "commands/dbcommands.h"
+#include "commands/sequence.h"
+#include "distributed/argutils.h"
+#include "distributed/background_jobs.h"
+#include "distributed/citus_safe_lib.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/colocation_utils.h"
+#include "distributed/connection_management.h"
+#include "distributed/enterprise.h"
+#include "distributed/hash_helpers.h"
+#include "distributed/listutils.h"
+#include "distributed/lock_graph.h"
+#include "distributed/coordinator_protocol.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/metadata_utility.h"
+#include "distributed/multi_logical_replication.h"
+#include "distributed/multi_progress.h"
+#include "distributed/multi_server_executor.h"
+#include "distributed/pg_dist_rebalance_strategy.h"
+#include "distributed/pg_dist_shard.h"
+#include "distributed/reference_table_utils.h"
+#include "distributed/remote_commands.h"
+#include "distributed/resource_lock.h"
+#include "distributed/shard_rebalancer.h"
+#include "distributed/shard_cleaner.h"
+#include "distributed/shard_transfer.h"
+#include "distributed/tuplestore.h"
+#include "distributed/utils/array_type.h"
+#include "distributed/worker_protocol.h"
+#include "nodes/pg_list.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+#include "postmaster/postmaster.h"
+#include "storage/lmgr.h"
+#include "utils/builtins.h"
+#include "utils/fmgroids.h"
+#include "utils/json.h"
+#include "utils/lsyscache.h"
+#include "utils/memutils.h"
+#include "utils/pg_lsn.h"
+#include "utils/syscache.h"
+#include "common/hashfn.h"
+#include "utils/varlena.h"
+#include "utils/guc_tables.h"
+#include "distributed/commands/utility_hook.h"
+
+#define MAX_SHARD_SIZE  1
+PG_FUNCTION_INFO_V1(citus_auto_shard_split_start); 
+
+void Citus_Table_List(void)
```

Comment:
make sure you follow https://github.com/citusdata/citus/blob/main/CONTRIBUTING.md#following-our-coding-conventions